selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/RenameWorkspaceTest.java

```diff
@@ -65,45 +61,39 @@ public void renameNameWorkspaceTest() throws IOException {
     dashboardWorkspace.selectTabInWorspaceMenu(DashboardWorkspace.TabNames.OVERVIEW);
-    dashboardWorkspace.enterNameWorkspace(CHANGE_WORKSPACE_NAME);
-    dashboardWorkspace.clickOnSaveBtn();
-    dashboardWorkspace.checkStateOfWorkspace(StateWorkspace.STOPPING);
 
-    // This temporary solution for detect problem with this test
-    // we will make screenshot every 5 ms for understanding problem
-    screenshot(StateWorkspace.STOPPING.getStatus());
+    // type name with 1 characters and check error message that this name is too short
+    dashboardWorkspace.enterNameWorkspace("w");
+    assertTrue(dashboardWorkspace.isWorkspaceNameErrorMessageEquals(WS_NAME_TOO_SHORT));
+    dashboardWorkspace.clickOnCancelBtn();
+    dashboardWorkspace.checkNameWorkspace(workspaceName);
 
-    int i = 1;
-    while (!checkStateOfWorkspaceIsStarting() && i < 120) {
-      screenshot(StateWorkspace.STARTING.getStatus() + i);
-      WaitUtils.sleepQuietly(500, TimeUnit.MILLISECONDS);
-      i++;
-    }
+    // type name with 101 characters and check error message that this name is too long
+    dashboardWorkspace.enterNameWorkspace(MAX_WORKSPACE_NAME + "1");
+    assertTrue(dashboardWorkspace.isWorkspaceNameErrorMessageEquals(WS_NAME_TOO_LONG));
+    dashboardWorkspace.clickOnCancelBtn();
+    dashboardWorkspace.checkNameWorkspace(workspaceName);
 
-    dashboardWorkspace.checkStateOfWorkspace(StateWorkspace.RUNNING);
-    dashboard.waitNotificationMessage("Workspace updated");
-    dashboard.waitNotificationIsClosed();
-    dashboardWorkspace.checkNameWorkspace(CHANGE_WORKSPACE_NAME);
+    // type a name with min possible size and check that the workspace renamed
+    renameWorkspace(MIN_WORKSPACE_NAME);
+
+    // type a name with max possible size and check that the workspace renamed
+    renameWorkspace(MAX_WORKSPACE_NAME);
   }
 
-  private void screenshot(String m) throws IOException {
-    byte[] data = ((SeleniumWebDriver) dashboard.driver()).getScreenshotAs(OutputType.BYTES);
-    Path screenshot =
-        Paths.get(Paths.get("target/screenshots").toString(), "RenameWorkspaceTest_" + m + ".png");
-    Files.createDirectories(screenshot.getParent());
-    Files.copy(new ByteArrayInputStream(data), screenshot);
+  private void renameWorkspace(String name) {
+    dashboardWorkspace.enterNameWorkspace(name);
+    assertFalse(dashboardWorkspace.isWorkspaceNameErrorMessageEquals(WS_NAME_TOO_SHORT));
+    assertFalse(dashboardWorkspace.isWorkspaceNameErrorMessageEquals(WS_NAME_TOO_LONG));
+    clickOnSaveButton();
+    dashboardWorkspace.checkNameWorkspace(name);
   }
 
-  public boolean checkStateOfWorkspaceIsStarting() {
-    try {
-      dashboard
-          .driver()
-          .findElement(
-              By.xpath(
-                  format("//div[contains(@class, 'workspace-status')]/span[text()='starting']")));
-      return true;
-    } catch (NoSuchElementException e) {
-      return false;
-    }
+  private void clickOnSaveButton() {
```

Comment:
It looks like a Dashboard Workspace page object specific method. So, I propose to move it into that class to comply [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) of OOP classes development.